### PR TITLE
Disable diags for specific server

### PR DESCRIPTION
**Pull request by andlrc**
Make it possible to disable language server features

Make it possible to disable diagnostics from a specific server.

### CODE REVIEW

File: autoload/lsp/diag.vim [hunk 1]
```diff
@@ -211,6 +211,11 @@ enddef
 # Notification: textDocument/publishDiagnostics
 # Param: PublishDiagnosticsParams
 export def DiagNotification(lspserver: dict<any>, uri: string, newDiags: list<dict<any>>): void
+  # Diagnostics are disabled for this server
+  if lspserver.features->has_key('diagnostics') && !lspserver.features.diagnostics
+    return
+  endif
+
```
Comment (andlrc) [inline autoload/lsp/diag.vim:218]: I don't think there is any way in the specifications to tell a language server NOT to send diagnostics, so they are ignored when they are received instead.

File: doc/lsp.txt [hunk 2]
```diff
@@ -1411,5 +1398,65 @@ typescript-language-server for everything else: >
 		}
 	])
 <
+Another way to do it is to disable the unwanted features, for exmaple if you
```
Comment (yegappan) [inline doc/lsp.txt:1401]: Typo: "exmaple" -> "example"

File: doc/lsp.txt [hunk 3]
```diff
@@ -1411,5 +1398,65 @@ typescript-language-server for everything else: >
...
+			path: '/usr/local/bin/typescript-language-server',
+			args: ['--stdio'],
+			features: {
+				diagnostics: false
+			}
+		},
+	])
+<
+==============================================================================
+17. Language Server Featured			*lsp-features*
```
Comment (yegappan) [inline doc/lsp.txt:1419]: Typo: "Featured" -> "Features"

File: doc/lsp.txt [hunk 4]
```diff
@@ -1411,5 +1398,65 @@ typescript-language-server for everything else: >
...
+
+By proving the configuration |lsp-cfg-features| it's possible specify which
```
Comment (yegappan) [inline doc/lsp.txt:1421]: Typo: "proving" -> "providing"
"possible specify" -> "possible to specify"

**Review by yegappan — CHANGES_REQUESTED** [review]

File: doc/lsp.txt (hunk 4 above)
Comment (andlrc) [reply doc/lsp.txt]: wow it's almost impressive how many typos I managed to get in such a small change 😅

**Pull request merged**